### ISSUE TITLE
TypeError: get_context() got an unexpected keyword argument 'renderer' on Django 1.11.3

### ISSUE THREAD

**Issue by Gwildor** [jazzband/django-floppyforms#192]
The error in the title is raised for Floppyforms widgets on Django 1.11.3 because of the change in [this ticket](https://code.djangoproject.com/ticket/28265) (code [here](https://github.com/django/django/pull/8584)). The solution for now is to stay on 1.11.2 or lower until this is fixed in Floppyforms.

**Comment by lzanuz** [comment]
I confirm the problem on django 1.11.4
floppyforms/widgets.py in render(self, name, value, attrs, **kwargs)
get_context() got an unexpected keyword argument 'renderer'
on django==1.11.2 works fine!

**Comment by loge-gh** [comment]
+1

**Comment by bnisevic** [comment]
Btw, we don't use floppyforms anymore.

**Comment by johnnyrockets** [comment]
+1

**Comment by lzanuz** [comment]
Nice job @johnnyrockets 
@gregmuellegger - Please fix and upload a new version on pip repository.

**Comment by Jasper-Koops** [comment]
Has the issue been fixed?

**Comment by amenoni** [comment]
+1

**Comment by eyesee1** [comment]
+1

**Comment by skar3** [comment]
+1

**Comment by oTree-org** [comment]
+1

**Comment by fronbasal** [comment]
+1

**Comment by brent960** [comment]
+1

**Comment by chrnolte** [comment]
+1

**Comment by intelef** [comment]
+1

**Comment by joaobarcia** [comment]
Is there any chance of including the fix in a new release? Or is there any proposed official temp workaround?

I am now upgrading an older Django to 2.0. The project makes use of floppyforms to a great extent, so this would really be a requirement.

THanks

**Comment by skar3** [comment]
Please!

**Comment by manojrege** [comment]
Fixed here: https://github.com/gregmuellegger/django-floppyforms/pull/193

**Comment by lokkomokko** [comment]
I have simillar problem on django 2.0.2 , please help somebody

**Comment by davewilliamstx** [comment]
Fix 193 from @manojrege works for me!

**Comment by alemangui** [comment]
@lokkomokko try @manojrege's fix, it worked for me on django 2.0.2

**Comment by jojjo64** [comment]
+1

**Comment by nnamdiib** [comment]
@manojrege Thanks man!

**Comment by skar3** [comment]
Guys, after the @manojrege fix, how can i add additional css class using attrs? it worked before:
Example:
forms.DateInput(attrs={'class': 'form-control'})

**Comment by amirasaad** [comment]
+1

**Comment by aquilante** [comment]
I know that this is an old issue but to circumvent the problem i **monkeypatched** like this:
```
from floppyforms import widgets
old_render = widgets.Input.render

def patched_render(self, name, value, attrs=None, **kwargs):
    renderer = kwargs.pop('renderer', None)
    if renderer:
        from floppyforms.compat import MULTIVALUE_DICT_TYPES, flatten_contexts
        template_name = kwargs.pop('template_name', None)
        if template_name is None:
            template_name = self.template_name
        context = self.get_context(name, value, attrs=attrs or {}, **kwargs)
        context = flatten_contexts(self.context_instance, context)
        return renderer.render(template_name, context)        
    else:
        return old_render(self, name, value, attrs, **kwargs)

widgets.Input.render = patched_render
```
I know this is ugly but as a quick fix it does its job.

There is a similar issues with passing extra `attrs` for the widgets.

**Comment by ramonakira** [comment]
We use django-floppyforms in a few of our projects and would greatly appreciate a fix that works with Python 3 and Django 2.2. If you need any help updating this project I offer my assistance.

**Comment by rtpg** [comment]
I believe that all of y'all's issues should be fixed with the latest release (`1.8.0`).

If you hit any issues still with that release, please re-open this issue/create a new issue and we'll see what we can do
